gateway/exporters/fluentd/fluentd.go

```diff
@@ -0,0 +1,229 @@
+package fluentd
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/fluent/fluent-logger-golang/fluent"
+	"github.com/openconfig/gnmi-gateway/gateway/configuration"
+	"github.com/openconfig/gnmi-gateway/gateway/connections"
+	"github.com/openconfig/gnmi-gateway/gateway/exporters"
+	"github.com/openconfig/gnmi-gateway/gateway/utils"
+	"github.com/openconfig/gnmi/ctree"
+	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+const Name = "fluentd"
+const MetricType = "log"
+const LogType = "log"
+const LoggedMetricType = "loggedMetric"
+
+var _ exporters.Exporter = new(FluentdExporter)
+
+type Log struct {
+	Namespace     string            `msg:"eventNamespace"`
+	Event         string            `msg:"eventName"`
+	GnmiPath      string            `msg:"gnmiPath"`
+	SubscribePath string            `msg:"subscribedPath"`
+	Value         interface{}       `msg:"value"`
+	Meta          map[string]string `msg:"meta"`
+	FabricID      string            `msg:"fabricId"`
+	RackID        string            `msg:"rackId"`
+	DeviceID      string            `msg:"deviceId"`
+	ExtensionID   string            `msg:"extensionId"`
+	DeviceName    string            `msg:"deviceName"`
+	Timestamp     string            `msg:"gnmiTimestamp"`
+}
+
+func init() {
+	exporters.Register(Name, NewFluentdExporter)
+}
+
+func NewFluentdExporter(config *configuration.GatewayConfig) exporters.Exporter {
+	return &FluentdExporter{
+		config: config,
+	}
+}
+
+type FluentdExporter struct {
+	config       *configuration.GatewayConfig
+	connMgr      *connections.ConnectionManager
+	fluentLogger *fluent.Fluent
+}
+
+func (e *FluentdExporter) Name() string {
+	return Name
+}
+
+func (e *FluentdExporter) Export(leaf *ctree.Leaf) {
+	notification := leaf.Value().(*gnmipb.Notification)
+
+	for _, update := range notification.Update {
+		log := Log{}
+
+		value, valid := utils.GetValues(update.Val)
+
+		if !valid {
+			continue
+		}
+
+		log.Value = value
+
+		notificationPath := utils.GetTrimmedPath(notification.Prefix, update.Path)
+
+		elems, keys, err := extractPrefixAndPathKeys(notification.GetPrefix(), update.GetPath())
+		if err != nil {
+			e.config.Log.Info().Msg(fmt.Sprintf("Failed to extract notificationPath or keys: %s", err))
+		}
+
+		if log.Event == "" {
```

Comment:
we're initializing ``log`` at line 65. Can this be anything other than an empty string? I think we can drop this check and have the code a little less nested.